### PR TITLE
[RLlib] Cleanup examples folder 18: Add example script for offline RL (BC) training on single-agent, while evaluating w/ multi-agent setup.

### PR DESCRIPTION
**Pull request by sven1977**
Cleanup examples folder 18: Add example script for offline RL (BC) training on single-agent, while evaluating w/ multi-agent setup.

* Some minor bug fixes were necessary to make this new example run.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/bc/bc.py [hunk 1]
```diff
@@ -166,34 +168,51 @@ def training_step(self) -> ResultDict:
                         max_env_steps=self.config.train_batch_size,
                     )
 
-                train_batch = train_batch.as_multi_agent()
-                self._counters[NUM_AGENT_STEPS_SAMPLED] += train_batch.agent_steps()
-                self._counters[NUM_ENV_STEPS_SAMPLED] += train_batch.env_steps()
+                # TODO (sven): Use metrics API as soon as we moved to new API stack
```
Comment (simonsays1980) [inline rllib/algorithms/bc/bc.py:171]: Is this one not using the `MetricsLogger`, yet? I use it in my overhaul of offline RL

File: rllib/examples/offline_rl/pretrain_bc_single_agent_evaluate_as_multi_agent.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
+# @HybridAPIStack
+
+"""Example showing how to train a (SA) BC RLModule while evaluating in a MA setup.
+
+Here, SA=single-agent and MA=multi-agent.
+
+Note that the BC Algorithm - by default - runs on the hybrid API stack, using RLModules,
+but not EnvRunners or ConnectorV2s yet.
```
Comment (simonsays1980) [inline rllib/examples/offline_rl/pretrain_bc_single_agent_evaluate_as_multi_agent.py:8]: Probably adding here that is specifically doe snot use `SingleAgentEpisode/MultiAgentEpisode`?

File: rllib/examples/offline_rl/pretrain_bc_single_agent_evaluate_as_multi_agent.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    dummy_env = gym.make("CartPole-v1")
+
+    offline_file = str(
+        Path(os.path.join("../../", "tests/data/cartpole/large.json")).resolve()
+    )
+
+    base_config = (
+        BCConfig()
+        .environment(
+            observation_space=dummy_env.observation_space,
```
Comment (simonsays1980) [inline rllib/examples/offline_rl/pretrain_bc_single_agent_evaluate_as_multi_agent.py:96]: Can we give a quick note, why in this case the user needs to provide the spaces?

**Review by simonsays1980 — APPROVED** [review]
LGTM. Some comments.

File: rllib/algorithms/bc/bc.py (hunk 1 above)
Comment (sven1977) [reply rllib/algorithms/bc/bc.py:171]: Yes, you should. All good there. :)

But the hybrid API stack still goes through the `summarize_episodes` utility inside `algorithm.py`, which operates on the old `RolloutMetrics` objects returned by RolloutWorkers.

**Pull request merged**